navigator/backend/src/test/scala/com/digitalasset/navigator/backend/IntegrationTest.scala

```diff
@@ -60,41 +68,70 @@ class IntegrationTest
         .transform(_ => ta)
     }
   }
 
-  def getResponseDataBytes(resp: HttpResponse): Future[String] = {
-    val fb = resp.entity.dataBytes.runFold(ByteString.empty)((b, a) => b ++ a).map(_.utf8String)
-    fb
+  def getResponseDataBytes(resp: HttpResponse): Future[String] =
+    resp.entity.dataBytes.runFold(ByteString.empty)((b, a) => b ++ a).map(_.utf8String)
+
+  "Navigator (parties)" - {
+    "picks up newly allocated parties" in withNavigator(enableUserManagement = false) {
+      case (uri, client) =>
+        for {
+          resp <- Http().singleRequest(HttpRequest(uri = uri.withPath(Uri.Path("/api/session/"))))
+          respBody <- getResponseDataBytes(resp)
+          _ = respBody shouldBe """{"method":{"type":"select","users":[]},"type":"sign-in"}"""
+          _ = resp.status shouldBe StatusCodes.OK
+          _ <- client.partyManagementClient
+            .allocateParty(hint = None, displayName = Some("display-name"))
+          _ <- RetryStrategy.constant(20, 1.second) { case (run @ _, duration @ _) =>
+            for {
+              resp <- Http().singleRequest(
+                HttpRequest(uri = uri.withPath(Uri.Path("/api/session/")))
+              )
+              respBody <- getResponseDataBytes(resp)
+            } yield {
+              respBody shouldBe """{"method":{"type":"select","users":["display-name"]},"type":"sign-in"}"""
+            }
+          }
+        } yield succeed
+    }
   }
 
-  "Navigator" - {
-    "picks up newly allocated parties" in withNavigator { case (uri, client) =>
-      for {
-        resp <- Http().singleRequest(HttpRequest(uri = uri.withPath(Uri.Path("/api/session/"))))
-        respBody <- getResponseDataBytes(resp)
-        _ = respBody shouldBe """{"method":{"type":"select","users":[]},"type":"sign-in"}"""
-        _ = resp.status shouldBe StatusCodes.OK
-        _ <- client.partyManagementClient
-          .allocateParty(hint = None, displayName = Some("display-name"))
-        _ <- RetryStrategy.constant(20, 1.second) { case (run @ _, duration @ _) =>
-          for {
-            resp <- Http().singleRequest(HttpRequest(uri = uri.withPath(Uri.Path("/api/session/"))))
-            respBody <- getResponseDataBytes(resp)
-          } yield {
-            respBody shouldBe """{"method":{"type":"select","users":["display-name"]},"type":"sign-in"}"""
+  "Navigator (users)" - {
+    "picks up newly created users" in withNavigator(enableUserManagement = true) {
+      case (uri, client) =>
+        for {
+          resp <- Http().singleRequest(HttpRequest(uri = uri.withPath(Uri.Path("/api/session/"))))
+          respBody <- getResponseDataBytes(resp)
+          _ = respBody shouldBe """{"method":{"type":"select","users":[]},"type":"sign-in"}"""
+          _ = resp.status shouldBe StatusCodes.OK
+          partyDetails <- client.partyManagementClient
+            .allocateParty(hint = None, displayName = Some("primary-party"))
+
+          _ <- client.userManagementClient
```

Comment:
maybe also add a test for having two users with the same primary party and a user with no primary party.